File: task-03/Go/Sub-Task4.go
package main
import (
    "io/ioutil"
    "strconv"
    "strings"
)

func main() {
    content, _ := ioutil.ReadFile("input.txt")
    n, _ := strconv.Atoi(strings.TrimSpace(string(content)))

    var output strings.Builder

    for i := 0; i < n; i++ {
        output.WriteString(strings.Repeat(" ", n-i-1) + strings.Repeat("*", 2*i+1) + "\n")
    }
    for i := n-2; i >= 0; i-- {
        output.WriteString(strings.Repeat(" ", n-i-1) + strings.Repeat("*", 2*i+1) + "\n")
    }

    ioutil.WriteFile("output.txt", []byte(output.String()), 0644)
}

File: task-03/Haskell/Sub-Task4.hs
import System.IO

main = do
    input <- readFile "input.txt"
    let n = read input :: Int
    let lines = [replicate (n-i-1) ' ' ++ replicate (2*i+1) '*' | i <- [0..n-1]]
            ++ [replicate (n-i-1) ' ' ++ replicate (2*i+1) '*' | i <- [n-2,n-3..0]]
    writeFile "output.txt" (unlines lines)
